### PR TITLE
add kernel spec to json kernelspec list output

### PR DESCRIPTION
**Pull request by emilienschultz**
pure json display of kernels informations

### CODE REVIEW

**Comment by Carreau** [comment]
+1

File: jupyter_client/kernelspecapp.py [hunk 1]
```diff
@@ -33,6 +33,7 @@ def _kernel_spec_manager_default(self):
 
     def start(self):
         specs = self.kernel_spec_manager.find_kernel_specs()
+        specs = {kname:{"dir":specs[kname],"info":self.kernel_spec_manager.get_kernel_spec(kname).to_dict()} for kname in specs}
```
Comment (takluyver) [inline jupyter_client/kernelspecapp.py]: We're not strict about keeping lines to 80 characters, but I think this is a bit too long. Can you split it onto two (or more) lines, and add a few spaces around the punctuation to make it  a bit more readable?

Comment (Carreau) [inline jupyter_client/kernelspecapp.py]: FYI, pep8 we where talking about this morning, that might be relevant: https://www.python.org/dev/peps/pep-0008/

**Comment by minrk** [comment]
To be more consistent with how we present this info elsewhere, can we call `info` `spec` and `dir` `resources_dir`?

**Comment by emilienschultz** [comment]
I made the changes suggested

**Comment by takluyver** [comment]
:+1: @minrk, I'll let you take another quick look before we merge.

**Comment by Carreau** [comment]
+1

**Comment by minrk** [comment]
Huzzah!

**Comment by Carreau** [comment]
images

**Comment by emilienschultz** [comment]
youhou !

**Comment by SylvainCorlay** [comment]
bravo @emilienschultz

**Pull request merged**